src/openspace/src/api.py

```diff
@@ -366,36 +369,18 @@ async def fun(*args):
                 _lib = '' if subPyLibrary == pyLibrary else libraryName + '.'
                 fullFunctionName = 'openspace.' + _lib + func['name']
 
-                if multiReturn:
-                    subPyLibrary[func['name']] = generateAsyncMultiRetFunction(fullFunctionName)
-                else:
-                    subPyLibrary[func['name']] = generateAsyncSingleRetFunction(fullFunctionName)
-
-        return self.toNamedTuple(pyLibrary, libraryName)
-
-    def toNamedTuple(self, content: dict, name: str = "namedtuple") -> NamedTuple:
-        """ Recursively converts a `dictionary` to a `namedtuple`. """
+                lua_call = partial(async_lua_call, fullFunctionName)
+                if wrapper is not None:
+                    lua_call = partial(wrapper, lua_call)
+                subPyLibrary[func['name']] = lua_call
 
-        T = namedtuple(name, content.keys())
-        values = []
-        for k, v in content.items():
-            if isinstance(v, dict):
-                values.append(self.toNamedTuple(v, k))
-            else:
-                values.append(v)
-
-        return T(*values)
+        return toNamedTuple(pyLibrary, libraryName)
 
-    async def singleReturnLibrary(self):
+    async def singleReturnLibrary(self, wrapper=None):
         """ Get an object representing the OpenSpace lua library. \n
+        (deprecated. Use library() instead)
+        :param wrapper: if set, wraps all API calls (may be used to make them synchronous)
```

Comment:
We could wrap this in the `@deprecated` decorator, however, this would bump Python requirements to 3.13.3. Thoughts @alexanderbock ?